File: core/src/main/java/com/makeuseof/muocore/models/PublisherPost.java
package com.makeuseof.muocore.models;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonSetter;

import java.text.SimpleDateFormat;
import java.util.Date;

/**
 * Created by abaikirov on 10/3/17.
 */


@JsonIgnoreProperties(ignoreUnknown = true)

public class PublisherPost {

    private static SimpleDateFormat date_format = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSS'Z'");

    @JsonProperty("ID")
    public int id;

    public Date date;

    @JsonSetter("post_date")
    public void setPostDate(final String postDate){
        try{
            this.date = date_format.parse(postDate);
        } catch (Exception e){
            e.printStackTrace();
        }
    }

    @JsonProperty("post_title")
    public String title;

    @JsonProperty("author")
    public String author;

    @JsonProperty("html")
    public String html;

    @JsonProperty("url")
    public String url;

    @JsonProperty("featured_image")
    public PublisherImage image;

}

File: app/src/main/kotlin/com/makeuseof/cryptocurrency/data/post/model/PublisherPost.kt
package com.makeuseof.cryptocurrency.data.post.model

import com.squareup.moshi.Json

import java.text.SimpleDateFormat

/**
 * Created by askar on 11/19/18
 * with Android Studio
 */

data class PublisherPost(
        @Json(name = "ID") var id: Int,
        @Json(name = "post_date") var date: String?,
        @Json(name = "post_title") var title: String?,
        @Json(name = "author") var author: String?,
        @Json(name = "html") var html: String?,
        @Json(name = "url") var url: String?,
        @Json(name = "featured_image") var image: PublisherImage?
) {

    companion object {
        private val date_format = SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSS'Z'")
    }
}
